xcdat/regridder/xesmf.py

```diff
@@ -135,69 +143,34 @@ def __init__(
         self._extrap_dist_exponent = extrap_dist_exponent
         self._extrap_num_src_pnts = extrap_num_src_pnts
         self._ignore_degenerate = ignore_degenerate
-        self._regridder: xe.XESMFRegridder = None
         self._extra_options = options
 
     def vertical(self, data_var: str, ds: xr.Dataset) -> xr.Dataset:
         """Placeholder for base class."""
         raise NotImplementedError()
 
     def horizontal(self, data_var: str, ds: xr.Dataset) -> xr.Dataset:
-        """Regrid ``data_var`` in ``ds`` to output grid.
-
-        Parameters
-        ----------
-        data_var : str
-            The name of the data variable inside the dataset to regrid.
-        ds : xr.Dataset
-            The dataset containing ``data_var``.
-
-        Returns
-        -------
-        xr.Dataset
-            Dataset with variable on the destination grid.
-
-        Raises
-        ------
-        KeyError
-            If data variable does not exist in the Dataset.
-
-        Examples
-        --------
-
-        Create output grid:
-
-        >>> output_grid = xcdat.create_gaussian_grid(32)
-
-        Create regridder:
-
-        >>> regridder = xesmf.XESMFRegridder(ds, output_grid, method="bilinear")
-
-        Regrid data:
-
-        >>> data_new_grid = regridder.horizontal("ts", ds)
-        """
+        """See documentation in :py:func:`xcdat.regridder.xesmf.XESMFRegridder`"""
```

Comment:
Is this the main regrid method used in the example notebooks? Or just a function that is wrapped by something else? If it is the method users would interact with, I think we should extend the documentation.

Reply:
I have the same question while going through the docs. To me, using `RegridderAccessor` class might be the main method users would use? Using individual regridder class is also supported. @jasonb5 let me know if this is the case. I was going to make some doc changes anyway, so if it is okay with you, I can put together a PR to address Steve's review.

Reply:
@pochedls @chengzhuzhang The intended usage is for users to call `RegridderAccessor.horizontal` and `RegridderAccessor.vertical`, these call the `horizontal` and `vertical` methods on the regridder clases, which are intended to stay hidden from the user.

**_There is a note that goes along with this._**

I had originally designed the API to be used in two ways.
1. A single instance where a user calls `ds.regridder.horizontal(...)`.
2. A multiple instance where a use might create a `regridder` from some input grid and call the `horizonal` or `vertical` methods multiple times. The intention here was the weights would be calculated once and used to regrid multiple input datasets. See the example below.

```python
regridder = xcdat.regridder.XESMFRegridder(input_grid, ...)

... = regridder.horizontal(ds1)
... = regridder.horizontal(ds2)
...
```

In hindsight this was not a great API design, an the reason why `horizontal` and `vertical` methods on the regridders will stay hidden from the user.

I have another PR coming that is going to address weights and bring this API more in line with the others in this area. This will also provide the feature of sharing weights to regrid multiple files.
```python
weights = ...
ds.regridder.horizontal(..., weights=weights) # user defined weights

output = ds.regridder.horizontal('tas', ..., keep_weights=True)
weights = output['regrid_weights'] # 
```
@chengzhuzhang You can make the changes directly to my branch.

Reply:
@jasonb5 Thank you for the clarification and note. I will go ahead the make changes to the notebooks in your branch. The new API change that allows keeping and sharing weights sounds good to me.  Though @tomvothecoder is aiming for a release next week, we can discuss tomorrow if the new change can make the new release.